actions/storage.js

```diff
@@ -80,13 +80,29 @@ export function actionOnItemsWithPrefix(prefix, action) {
     };
 }
 
-export function storageRehydrate(incoming) {
-    return async (dispatch, persistor) => {
+export function storageRehydrate(incoming, persistor) {
+    return async (dispatch, getState) => {
+        const state = getState();
         persistor.pause();
         Object.keys(incoming).forEach((key) => {
             const storage = {};
             try {
-                storage[key] = JSON.parse(incoming[key]);
+                const value = JSON.parse(incoming[key]);
+                if (typeof state.storage.storage[key] === 'undefined') {
+                    if (typeof value.timestamp === 'undefined') {
+                        storage[key] = {value, timestamp: new Date()};
+                    } else {
+                        storage[key] = value;
```

Comment:
Why do we store both the plain value and the wrapped value, instead of always wrapping but flagging via the undefined `timestamp` instead? It seems like this adds unnecessary complexity in the reducers below to handle both cases.

Reply:
The explanation is the backward compatibility. For 4.7 and before, the storage always store the data using the value. So, the user can still have old data format in his storage, this support old data format, and new data format, always transforming it to the new data format. Other approach is do some kind of data migration for this, and leave the code simpler in that part.

Reply:
But doesn't this method rehydrate everything from the store? That is, we can effectively upgrade the schema right here? I get that we have to handle backwards compatibility somewhere, but it's unclear why we have to do it here /and/ in the reducers. Or am I misunderstanding the rehydration step?

Reply:
I think you are right, the entry points is the rehydratation, so, the migration must go there, every where else, doesn't need migration at all. I going to review it.